projects/js-packages/ai-client/src/hooks/use-ai-suggestions/index.ts

```diff
@@ -0,0 +1,365 @@
+	 */
+	autoRequest?: boolean;
+
+	/**
+	 * AskQuestion options.
+	 */
+	askQuestionOptions?: AskQuestionOptionsArgProps;
+
+	/*
+	 * onSuggestion callback.
+	 */
+	onSuggestion?: ( suggestion: string ) => void;
+
+	/*
+	 * onDone callback.
+	 */
+	onDone?: ( content: string ) => void;
+
+	/*
+	 * onError callback.
+	 */
+	onError?: ( error: SuggestionErrorProps ) => void;
+};
+
+export type RequestingStateProp = 'init' | 'requesting' | 'suggesting' | 'done' | 'error';
+
+type useAiSuggestionsProps = {
+	/*
+	 * The suggestion.
+	 */
+	suggestion: string;
+
+	/*
+	 * The error.
+	 */
+	error: SuggestionErrorProps | undefined;
+
+	/*
+	 * The post ID.
+	 */
+	postId: number;
+
+	/*
+	 * Whether the request is in progress.
+	 */
+	requestingState: RequestingStateProp;
+
+	/*
+	 * The event source.
+	 */
+	eventSource: SuggestionsEventSource | undefined;
+
+	/*
+	 * The request handler.
+	 */
+	request: ( prompt: Array< PromptItemProps > ) => Promise< void >;
+};
+
+const debug = debugFactory( 'jetpack-ai-client:use-suggestion' );
+
+/**
+ * Get the error data for a given error code.
+ *
+ * @param {SuggestionErrorCode} errorCode - The error code.
+ * @returns {SuggestionErrorProps}          The error data.
+ */
+function getErrorData( errorCode: SuggestionErrorCode ): SuggestionErrorProps {
+	switch ( errorCode ) {
+		case ERROR_QUOTA_EXCEEDED:
+			return {
+				code: ERROR_QUOTA_EXCEEDED,
+				message: __( 'You have reached the limit of requests for this site.', 'jetpack-ai-client' ),
+				severity: 'info',
+			};
+		case ERROR_UNCLEAR_PROMPT:
+			return {
+				code: ERROR_UNCLEAR_PROMPT,
+				message: __( 'Your request was unclear. Mind trying again?', 'jetpack-ai-client' ),
+				severity: 'info',
+			};
+		case ERROR_SERVICE_UNAVAILABLE:
+			return {
+				code: ERROR_SERVICE_UNAVAILABLE,
+				message: __(
+					'Jetpack AI services are currently unavailable. Sorry for the inconvenience.',
+					'jetpack-ai-client'
+				),
+				severity: 'info',
+			};
+		case ERROR_MODERATION:
+			return {
+				code: ERROR_MODERATION,
+				message: __(
+					'This request has been flagged by our moderation system. Please try to rephrase it and try again.',
+					'jetpack-ai-client'
+				),
+				severity: 'info',
+			};
+		case ERROR_NETWORK:
+			return {
+				code: ERROR_NETWORK,
+				message: __(
+					'It was not possible to process your request. Mind trying again?',
+					'jetpack-ai-client'
+				),
+				severity: 'info',
+			};
+		default:
+			return {
+				code: ERROR_NETWORK,
+				message: __(
+					'It was not possible to process your request. Mind trying again?',
+					'jetpack-ai-client'
+				),
+				severity: 'info',
+			};
+	}
+}
+
+/**
+ * React custom hook to get suggestions from AI,
+ * by hitting the query endpoint.
+ *
+ * @param {useAiSuggestionsOptions} options - The options for the hook.
+ * @returns {useAiSuggestionsProps}           The props for the hook.
+ */
+export default function useAiSuggestions( {
+	prompt,
+	autoRequest = true,
```

Comment:
You said in the docs that it defaults to false:

```suggestion
	autoRequest = false,
```